tests/functional/conftest.py

```diff
@@ -44,13 +57,27 @@ def wait_for_maximum_list_length(
     ) -> None:
         """Wait for the list() length to be no greater than expected maximum or fail
         test if timeout is exceeded"""
-        for _ in range(max_iterations):
-            if len(rest_manager.list()) <= max_length:
+        logging.info(f"Checking {description!r} has no more than {max_length} items")
+        for count in range(max_iterations):
+            items = rest_manager.list()
+            if len(items) <= max_length:
                 break
+            logging.info(
+                f"Iteration: {count} Waiting for {description!r} items to be deleted: "
+                f"{[x.name for x in items]}"
+            )
             time.sleep(SLEEP_INTERVAL)
-        assert len(rest_manager.list()) <= max_length, (
-            f"Did not delete required items for {description}. "
-            f"Elapsed_time: {time.perf_counter() - start_time}"
+
+        items = rest_manager.list()
+        elapsed_time = time.perf_counter() - start_time
+        if len(items) > max_length:
+            logging.error(
+                f"Too many {description!r} items still remaining and timeout "
+                f"({elapsed_time}) exceeded: {[x.name for x in items]}"
+            )
+        assert len(items) <= max_length, (
+            f"Did not delete required items for {description!r}. "
+            f"Elapsed_time: {elapsed_time}"
```

Comment:
Looks like we're duplicating a bit here, but not sure if we can assert and log at the same time. Maybe just a plain assert is enough though.

Reply:
I made an error_message variable. Cleaner now.